### ISSUE TITLE
[Bug] The `shm_size` property in `resources` doesn't take effect

### ISSUE THREAD

**Issue by peterschmidt85** [dstackai/dstack#1006]
Steps to reproduce:

1. Define `.dstack.yml`

```yaml
type: dev-environment

python: 3.11

ide: vscode

resources:
  shm_size: 4GB
```

2. `dstack run . -b aws`

3. Inside the dev environment, run `df -h /dev/shm`

It prints:

```
Filesystem      Size  Used Avail Use% Mounted on
shm              64M     0   64M   0% /dev/shm
```

> [!NOTE]
> If I `ssh <run name>host`, and run `docker inspect <container id> | grep -i shm`, it prints:
> ```
> "ShmSize": 67108864,
> ```

**Comment by peterschmidt85** [comment]
This feature used to work but when we introduced `dstack-shim`, we lost this functionality. Basically, we simply need to pass `shm_size` (if set) to `docker run` where we run the container.

**Comment by peterschmidt85** [comment]
Relevant code:

- [runner/internal/shim/docker.go:164](https://github.com/dstackai/dstack/blob/49bbf580d63345df532aa084b4b10976019b201f/runner/internal/shim/docker.go#L164-L163)
- [runner/internal/shim/api/http.go:37](https://github.com/dstackai/dstack/blob/master/runner/internal/shim/api/http.go#L37)

**Comment by peterschmidt85** [comment]
Released in `0.16.3` ([Discord link](https://discord.com/channels/1106906313969123368/1163422692226113538/1217800685752422450))
